packages/core/src/lib/markdown-it/plugins/custom-component/customComponentPlugin.js

```diff
@@ -0,0 +1,19 @@
+const customComponentHtmlBlockRule = require('./htmlBlockRule');
+const htmlInlineRule = require('./htmlInlineRule');
+
+/**
+ * Replacing the default htmlBlock rule to allow using custom components in markdown
+ */
+const customComponentPlugin = (md) => {
+  /*
+   * Note that html_block is to be replaced by markdown-it-escape-special-tags.
+   * Thus, we have to push a new rule after the original html_block rule instead.
+   */
+  md.block.ruler.after('html_block', 'custom_component_html_block', customComponentHtmlBlockRule, {
```

Comment:
Let's combine the two patches, since both relate to the idea of custom components and edit the same markdown rule.

(could move this over to the `markdown-it/patches/` directory instead as well since it modifies markdown-it default behaviour)

Reply:
@ang-zeyu Hmm now that you mention it, I agree that they do relate to the idea of custom components. I've made the changes accordingly. Let me know what you think of it :-)